src/Features/CSharp/Portable/Completion/KeywordRecommenders/StackAllocKeywordRecommender.cs

```diff
@@ -16,20 +16,41 @@ public StackAllocKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            // type t = |
-            var token = context.TargetToken;
-            if (token.IsUnsafeContext())
+            var node = context.TargetToken.Parent;
+
+            if (node == null)
+            {
+                return false;
+            }
+
+            if (node.IsKind(SyntaxKind.ParenthesizedExpression) || node.IsKind(SyntaxKind.CastExpression))
+            {
+                node = node.Parent;
+            }
+
+            while (node.IsKind(SyntaxKind.ConditionalExpression))
```

Comment:
should only do this if the would prefer this only happen if you were on the ? or :